messaging/src/main/java/io/lantern/messaging/CryptoWorker.kt

```diff
@@ -792,7 +785,13 @@ internal class CryptoWorker(
                     }
                 else ->
                     if (!tx.contains(senderId.directContactPath)) {
-                        throw UnknownSenderException(senderId)
+                        // automatically add unaccepted contact
+                        messaging.addOrUpdateContact(
+                            senderId.directContactId,
+                            "",
+                            Model.ContactSource.UNSOLICITED,
+                            initialVerificationLevel = Model.VerificationLevel.UNACCEPTED
+                        )
```

Comment:
does this mean that "in theory" my inbox can get filled with unsolicited contact requests?

Reply:
Yes. The good news is that since we mark the contacts as such, we can sort these separately in the display.

In our UI, the main message list doesn't show all messages, it shows the most recent message from each contact. Messages from unaccepted contacts will sort separately at the bottom of the list.